backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/model/DocumentCollection.java

```diff
@@ -480,6 +480,14 @@ public void delete(final DocumentPath documentPath, final IDocumentChangesCollec
 		}
 
 		forRootDocumentWritable(rootDocumentPath, changesCollector, rootDocument -> {
+
+			final BooleanWithReason isDeleteForbidden = rootDocument.cannotBeDeleted();
```

Comment:
that's a sign that the "cannotBeDeleted" method might be called "isDeleteForbidden" :)